src/imagery/i.eodag/i.eodag.py

```diff
@@ -300,82 +396,199 @@ def no_fallback_search(search_parameters, provider):
+        return SearchResult([j for i in search_result for j in i])
     except Exception as e:
-        gs.verbose(e)
+        gs.debug(e)
         gs.fatal(_("Server error, please try again."))
 
 
-def create_products_dataframe(eo_products):
-    result_dict = {"id": [], "time": [], "cloudCover": [], "productType": []}
-    for product in eo_products:
-        for key in result_dict:
-            if key == "time":
-                if (
-                    "startTimeFromAscendingNode" in product.properties
-                    and product.properties["startTimeFromAscendingNode"] is not None
-                ):
+def list_products(products):
+    """Lists products on the Standard Output (Console) stream.
+
+    :param products: EO poducts to be listed
+    :type products: class:'eodag.api.search_result.SearchResult'
+    """
+    columns = ["id", "startTimeFromAscendingNode", "cloudCover", "productType"]
+    columns_NA = ["id_NA", "time_NA", "cloudCover_NA", "productType_NA"]
+    for product in products:
+        product_line = ""
+        for i, column in enumerate(columns):
+            product_attribute_value = product.properties[column]
+            # Display NA if not available
+            if product_attribute_value is None:
+                product_attribute_value = columns_NA[i]
+            else:
+                if column == "cloudCover":
+                    # Special formatting for cloud cover
+                    product_attribute_value = f"{product_attribute_value:2.0f}%"
+                elif column == "startTimeFromAscendingNode":
+                    # Special formatting for datetime
                     try:
-                        result_dict["time"].append(
-                            normalize_time(
-                                product.properties["startTimeFromAscendingNode"]
-                            )
+                        product_attribute_value = normalize_time(
+                            product_attribute_value
                         )
                     except:
-                        result_dict["time"].append(
-                            product.properties["startTimeFromAscendingNode"]
-                        )
-                else:
-                    result_dict["time"].append(None)
-            else:
-                if key in product.properties and product.properties[key] is not None:
-                    result_dict[key].append(product.properties[key])
-                else:
-                    result_dict[key].append(None)
+                        product_attribute_value = product.properties[column]
+            if i != 0:
+                product_line += " "
+            product_line += product_attribute_value
+        print(product_line)
 
-    df = pd.DataFrame().from_dict(result_dict)
-    return df
 
+def remove_duplicates(search_result):
+    """Removes duplicated products, in case a provider returns a product multiple time."""
```

Comment:
Does that happen in praxis? Or do you get duplicates from a search across different providers? I can imagine that UUIDs differ but scenes may be not (same scene file name, but different IDs in the different backends)... Is the latter handeled here?
```suggestion
    """Removes duplicated products, in case a provider returns a product multiple times."""
```

Reply:
Yes, especially cop_dataspace, not sure if there is other providers that has the same issue. But, cop_dataspace sometimes returns the same product twice (same scene name and same uuid). I have been trying to figure why it is happening, but no luck yet...